sisl/orbital.py

```diff
@@ -1130,3 +1131,54 @@ def __setstate__(self, d):
             self.__init__(d["name"], q0=d["q0"], tag=d["tag"])
         else:
             self.__init__(d["name"], (d["r"], d["f"]), q0=d["q0"], tag=d["tag"])
+
+
+@set_module("sisl")
+class HydrogenicOrbital(AtomicOrbital):
+    r""" A hydrogen-like atomic orbital defined by an effective atomic number Z in addition to the usual quantum numbers (n, l, m).
+
+    A hydrogenic atom (Hydrogen-like) is an atom with a single valence electron.
+
+    The returned orbital is properly normalized, see [1]_ for details.
+
+    References
+    ----------
+    .. [1] : https://en.wikipedia.org/wiki/Hydrogen-like_atom
+
+
+    Parameters
+    ----------
+    Z : float
+        effective atomic number
+    n : int
+        principal quantum number
+    l : int
+        angular momentum quantum number
+    m : int
+        magnetic quantum number
+    R : float, optional
+        max range of the constructed orbital
+
+    Examples
+    --------
+    >>> carbon_pz = HydrogenicOrbital(3.2, 2, 1, 0)
+
+    """
+
+    def __init__(self, Z, n, l, m, *args, **kwargs):
+
+        self._Z = Z
+
+        R = kwargs.get("R", 10.)
+        r = np.linspace(0, R, 1000)
+        z = 2 * Z / (n * a0("Ang"))
+        pref = (z ** 3 * factorial(n - l - 1) / (2 * n * factorial(n + l))) ** 0.5
+        L = eval_genlaguerre(n - l - 1, 2 * l + 1, z * r)
+        Rnl = pref * np.exp(-z * r / 2) * (z * r) ** l * L
+
+        super().__init__(n, l, m, (r, Rnl), **kwargs)
+
+
+    def copy(self):
+        """ Create an exact copy of this object """
+        return self.__class__(self._Z, self.n, self.l, self.m, self.q0, self.tag)
```

Comment:
this won't work, just saw... You are discarding *args* in the above, so you should pass q0 and tag as keyword arguments.

Actually you should remove `*args` from `__init__` to ensure that users don't pass something that is lost.

Reply:
I see. Better now?

Reply:
Yes, great!